sorc/fvcom_tools.fd/module_nwp.f90

```diff
@@ -23,44 +26,53 @@ module module_nwp
 
    private
    type :: nwp_type
-      character(len=6) :: datatype
-      integer :: numvar, xlat, xlon, xtime
-      integer :: i_mask, i_sst, i_ice, i_sfcT, i_iceT
-      character(len=20), allocatable :: varnames(:)
-      character(len=20), allocatable :: latname
-      character(len=20), allocatable :: lonname
-      character(len=20), allocatable :: dimnameEW
-      character(len=20), allocatable :: dimnameNS
-      character(len=20), allocatable :: dimnameTIME
-
-      real(r_kind), allocatable :: nwp_mask(:,:,:)
-      real(r_kind), allocatable :: nwp_sst(:,:,:)
-      real(r_kind), allocatable :: nwp_ice(:,:,:)
-      real(r_kind), allocatable :: nwp_sfcT(:,:,:)
-      real(r_kind), allocatable :: nwp_iceT(:,:,:) 
+      character(len=6) :: datatype !< Data type.
+      integer :: numvar !< Number of variabls.
+      integer :: xlat !< Number of latitudes.
+      integer :: xlon !< Number of longitudes.
+      integer :: xtime !< Number of times.
+      integer :: i_mask !< Is var visible (always 1).
+      integer :: i_sst !< Index of sst var.
+      integer :: i_ice !< Index of ice var.
+      integer :: i_sfcT !< Index of sst temp var.
+      integer :: i_iceT !< Index of ice temp var.
+      character(len=20), allocatable :: varnames(:) !< Variable names.
+      character(len=20), allocatable :: latname !< Latitude name.
+      character(len=20), allocatable :: lonname !< Longitude name.
+      character(len=20), allocatable :: dimnameEW !< East/West dimension name.
+      character(len=20), allocatable :: dimnameNS !< North/South dimension name.
+      character(len=20), allocatable :: dimnameTIME !< Time dimension name.
+
+      real(r_kind), allocatable :: nwp_mask(:,:,:) !< ???
+      real(r_kind), allocatable :: nwp_sst(:,:,:) !< ???
+      real(r_kind), allocatable :: nwp_ice(:,:,:) !< ???
+      real(r_kind), allocatable :: nwp_sfcT(:,:,:) !< ???
+      real(r_kind), allocatable :: nwp_iceT(:,:,:)  !< ???
    end type nwp_type
 
    type, extends(nwp_type) :: fcst_nwp
-      type(nwpbase), pointer :: head => NULL()
-      type(nwpbase), pointer :: tail => NULL()
+      type(nwpbase), pointer :: head => NULL() !< ???
```

Comment:
@dmwright526 Also what are head and tail for?

This comment from the code gives me some trepidation:

`!  HOW DOES THIS POINTER THING WORK?`

Reply:
@edwardhartnett These pointers are carryover from when I inherited the code from GSL's work with HRRR for a similar use... I am not sure with object based coding in Fortran if it needs to have parts initialized to gain access to the procedures within it...

Reply:
OK, I have added that info to the comments in the code.